query/src/storages/fuse/io/block_appender.rs

```diff
@@ -33,39 +35,79 @@ use crate::storages::fuse::statistics;
 use crate::storages::fuse::statistics::BlockMetaAccumulator;
 use crate::storages::fuse::statistics::StatisticsAccumulator;
 
-/// dummy struct, namespace placeholder
 pub struct BlockAppender;
 
 impl BlockAppender {
+    // A simple strategy of merging small blocks into larger ones:
+    // for each n successive data blocks in `blocks`, if the sum of their `memory_size` exceeds
+    //   `block_size_threshold`, they will be merged into one larger block.
+    //  NOTE:
+    //    - the max size of merge-block will be 2 * block_size_threshold
+    //    - for block that is larger than `block_size_threshold`, they will NOT be split
+    //        TODO handling split in table compact/optimize
+    pub(crate) fn reshape_blocks(
```

Comment:
I think it's better to sort the block in previous.

Then we can use this greedy merge in order.

Reply:
good idea!